lib/array.rb

```diff
@@ -0,0 +1,7 @@
+class Array
+  def fadi(&block)
+    element = self.first
+    self.shift if self.first == element
+    self.each{|n| element = block.call(element,n) }
```

Comment:
Space missing after comma.<br>Space missing to the left of {.<br>Space between { and | missing.